### PR TITLE
PLT-256 Added headers to getFile api call to trigger IE/Edge to download the file instead of just opening it

### PR DESCRIPTION
**Pull request by hmhealey**
IE and Edge don't respect the download attribute of an <a> tag so we need to set some header fields to get files to download properly.

Also updated the https://github.com/mssola/user_agent library so that it properly detects Edge.

### CODE REVIEW

**Comment by jwilander** [comment]
@hmhealey doesn't pass ESLint.

**Comment by hmhealey** [comment]
It's amazing how easy it is to fail ESLint when there's only 6 lines of Javascript changed. It should pass now.

**Comment by coreyhulen** [comment]
+1.  Also (0/5) if we should do this by detecting user/agent.

File: web/react/utils/utils.jsx [hunk 1]
```diff
@@ -986,6 +986,10 @@ export function getFileUrl(filename) {
     }
     url = getWindowLocationOrigin() + '/api/v1/files/get' + url;
 
+    if (download) {
+        url += '?download=true';
```
Comment (crspeller) [inline web/react/utils/utils.jsx]: Might want to restrict this one to IE. (is it even really necessary?)  As ? URLs won't be cached by many caching systems.

**Comment by hmhealey** [comment]
I'm going to rewrite this to not use a download flag and instead just check the user agent for IE/Edge when setting the headers.

**Comment by jwilander** [comment]
Closing since the new code is 100% different from that as the original PR. Please resubmit.

**Pull request closed without merging**